### PR TITLE
Purs 0.15

### PR DESCRIPTION
**Pull request by iostat**
NOTE: Before merging this, merge https://github.com/f-o-a-m/purescript-eth-core/pull/32 and tag it with a proper version so that packages.dhall doesn't have to refer to some arbitrary commit -- then update packages.dhall

NOTE: Before merging this, merge https://github.com/f-o-a-m/purescript-web3/pull/161 and tag it with a proper version so that packages.dhall doesn't have to refer to some arbitrary commit -- then update packages.dhall.
In the process of doing so, you'll also merge/tag/update spago.dhall-in-web3 https://github.com/f-o-a-m/purescript-eth-core/pull/32. So make sure you pull that tag in to this PR.

This one was the most straightfoward:
* Had to upgrade our `purescript-mkdirp` dependency to support Purs 0.15 -- this was merged/tagged as an executive decision and is now `v2.0.0`. Let's pray it works lmfao.
* Thanks to Martin's `yargs` -> `optparse` update, there was literally nothing to do here. Just upgrade the deps.
* I reenabled purs-tidy across tidy and ran it on the repo.

### CODE REVIEW

**Comment by martyall** [comment]
There are some deps in the packages.dhall file that I'm hoping might be in the 0.15 package set, like `tagged`, `errors`, `dodo-printer` etc. Basically all the things we didn't write. For the others that are already specified by `ps-web3`, I'm wondering if it makes more sense to just import the package set from `purescript-web3` directly and use them from there, rather than listing them explicitly here.

File: src/Data/GeneratorMain.purs [hunk 1]
```diff
@@ -62,9 +62,8 @@ generatorMain :: Effect Unit
 generatorMain = launchAff_ do
   (Args args) <- liftEffect $ execParser opts
   errs <- generatePS args
-  when (A.null errs)
-    $ liftEffect
-    $ exit 1
+  liftEffect <<< exit $
+    if (A.null errs) then 0 else 1
```
Comment (martyall) [inline src/Data/GeneratorMain.purs:66]: actually this just reminds me, we should really be logging these errors:

https://github.com/f-o-a-m/purescript-web3-generator/issues

Comment (iostat) [reply src/Data/GeneratorMain.purs:66]: We actually already log them, in `generatePS`: https://github.com/f-o-a-m/purescript-web3-generator/blob/purs-0.15/src/Data/CodeGen.purs#L68-L72

File: example/src/Main.purs [hunk 2]
```diff
@@ -1,5 +1,6 @@
-module Generator where
```
Comment (martyall) [inline example/src/Main.purs:1]: You might need to change the README if you change  the module name here.

**Pull request merged**